src/main/java/net/syscon/elite/service/BedAssignmentHistoryService.java

```diff
@@ -15,11 +17,13 @@
 
     private final BedAssignmentHistoriesRepository repository;
 
-    public BedAssignmentHistoryService(BedAssignmentHistoriesRepository repository) {
+    public BedAssignmentHistoryService(final BedAssignmentHistoriesRepository repository) {
         this.repository = repository;
     }
 
-    public void add(Long bookingId, Long livingUnitId, String reasonCode, LocalDateTime time) {
+    @VerifyBookingAccess
+    @HasWriteScope
+    public void add(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime time) {
         final var maxSequence = repository.getMaxSeqForBookingId(bookingId);
         final var bookingAndSequence = new BedAssignmentHistory.BedAssignmentHistoryPK(bookingId, maxSequence + 1);
```

Comment:
Is there no underlying (Oracle) sequence for this?  or is this how old NOMIS does it here?

Reply:
Yes second that - check for sequence

Reply:
Old Nomis grabs the latest sequence and adds one - so I duplicated that code here.

Reply:
Sorry that code mentioned above is found in procedure `ins_off_bed_assg_hty`